includes/class-wc-shipping.php

```diff
@@ -165,7 +165,7 @@ public function load_shipping_methods( $package = array() ) {
 			$matched_zone_notice = sprintf( __( 'Customer matched zone "%s"', 'woocommerce' ), $shipping_zone->get_zone_name() );
 
 			// Debug output.
-			if ( $debug_mode && ! Constants::is_defined( 'WOOCOMMERCE_CHECKOUT' ) && ! Constants::is_defined( 'WC_DOING_AJAX' ) && ! wc_has_notice( $matched_zone_notice ) ) {
```

Comment:
This one seems like was made to test those messages, or there's another reason to remove checking for both constants?